src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/GetAzureSqlVM.cs

```diff
@@ -0,0 +1,127 @@
+        [Alias("SqlVMName")]
+        [ResourceNameCompleter("Microsoft.SqlVirtualMachine/SqlVirtualMachines", "ResourceGroupName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Resource id of the sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = ParameterSet.ResourceId,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = HelpMessages.SqlVMResourceId)]
+        [Alias("SqlVMId")]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Parse the parameters provided as input in order to obtain the name of the resource group and the sql virtual machine
+        /// </summary>
+        protected override void ParseInput()
+        {
+            if (ParameterSetName == ParameterSet.ResourceId)
+            {
+                if (!ValidateSqlVirtualMachineId(ResourceId))
+                {
+                    throw new PSArgumentException(
+                        string.Format("The sql virtual machine resource id is not well formatted"),
+                        "SqlVirtualMachine");
+                }
+                ResourceGroupName = GetResourceGroupNameFromId(ResourceId);
+                Name = GetResourceNameFromId(ResourceId);
+            }
+        }
+
+        /// <summary>
+        /// Gets one or more sql virtual machines.
+        /// </summary>
+        /// <returns></returns>
+        protected override IEnumerable<AzureSqlVMModel> GetEntity()
+        {
+            ICollection<AzureSqlVMModel> results = null;
+            if (ShouldGetByName(ResourceGroupName, Name))
+            {
+                results = new List<AzureSqlVMModel>();
+                results.Add(ModelAdapter.GetSqlVirtualMachine(ResourceGroupName, Name));
+            }
+            else if (ShouldListByResourceGroup(ResourceGroupName, Name))
+            {
+                results = ModelAdapter.ListSqlVirtualMachineByResourceGroup(ResourceGroupName);
+            }
+            else
+            {
+                results = ModelAdapter.ListSqlVirtualMachine();
+            }
+            return results;
+        }
+
+        /// <summary>
+        /// No user input to apply to model.
+        /// </summary>
+        /// <param name="model">The model to modify</param>
+        /// <returns>The input model</returns>
+        protected override IEnumerable<AzureSqlVMModel> ApplyUserInputToModel(IEnumerable<AzureSqlVMModel> model)
+        {
+            return model;
+        }
+
+        /// <summary>
+        /// No changes, thus nothing to persist.
+        /// </summary>
+        /// <param name="entity">The entity retrieved</param>
+        /// <returns>The unchanged entity</returns>
+        protected override IEnumerable<AzureSqlVMModel> PersistChanges(IEnumerable<AzureSqlVMModel> entity)
+        {
+            return entity;
```

Comment:
I see that at some places we are returning a List here and here we are just returning back the parameter. What is the difference?